membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -13,19 +13,31 @@ import models._
 import monitoring.Metrics
 import play.api.libs.concurrent.Execution.Implicits._
 import play.api.libs.json.Json
-import play.api.mvc.{Controller, Result}
+import play.api.mvc._
 import play.filters.cors.CORSActionBuilder
 
-import scala.concurrent.Future
+import scala.concurrent.{Future, Promise}
 import scalaz.std.scalaFuture._
 import scalaz.syntax.std.option._
 import scalaz.{EitherT, \/}
+import configuration.Config.authentication
 
 
 class AttributeController extends Controller with LazyLogging {
 
+  def signed(): ActionBuilder[Request] = new ActionBuilder[Request] {
+    def invokeBlock[A](request: Request[A], block: (Request[A]) => Future[Result]) = {
+      if (request.headers.get("authentication").contains(authentication.key)) {
```

Comment:
Thanks Adam.  I misread the line and thought that it said `Authorization`.  Agree on the capitalisation change, although the request headers map is case insensitive.